src/vcr_clj/core.clj

```diff
@@ -94,19 +96,22 @@
   (let [the-playbacker (playbacker cassette :key)
         redeffings
         (into {}
-              (for [{:keys [var arg-transformer arg-key-fn recordable?]
+              (for [{:keys [var arg-transformer arg-key-fn recordable? middleware]
                      :or {arg-transformer vector
                           arg-key-fn vector
+                          middleware (fn [f]
+                                       (fn [& args] (apply f args)))
                           recordable? (constantly true)}}
                     specs
                     :let [orig (deref var)
                           the-var-name (var-name var)
                           wrapped (fn [& args]
                                     (let [args* (apply arg-transformer args)]
                                       (if (apply recordable? args*)
-                                        (let [k (apply arg-key-fn args*)]
-                                          (:return (the-playbacker the-var-name k)))
-                                        (apply orig args*))))]]
+                                        (let [k (apply arg-key-fn args*)
+                                              res (:return (the-playbacker the-var-name k))]
+                                          (apply (middleware (fn [& args] res)) args*))
```

Comment:
I think the middleware could be applied to the real playback function rather than an artificial function via something like

``` clojure
(apply (middleware (fn [& args] (:return (the-playbacker the-var-name (apply arg-key-fn args))))))
```